sql/rename.go

```diff
@@ -68,3 +68,71 @@ func (p *planner) RenameDatabase(n *parser.RenameDatabase) (planNode, error) {
 
 	return &valuesNode{}, nil
 }
+
+// RenameTable renames the table.
+// Privileges: WRITE on database.
+//   Notes: postgres requires superuser, db owner, "CREATEDB", or table owner.
+//          mysql >= 5.1.23 requires ALTER and DROP on table.
+func (p *planner) RenameTable(n *parser.RenameTable) (planNode, error) {
+	if n.NewName == "" {
+		return nil, errEmptyTableName
+	}
+
+	if err := n.Name.NormalizeTableName(p.session.Database); err != nil {
+		return nil, err
+	}
+
+	dbDesc, err := p.getDatabaseDesc(n.Name.Database())
+	if err != nil {
+		return nil, err
+	}
+
+	tbKey := tableKey{dbDesc.ID, string(n.Name.Table())}.Key()
+
+	// Check if table exists
+	gr, err := p.txn.Get(tbKey)
+	if err != nil {
+		return nil, err
+	}
+	if !gr.Exists() {
+		if n.IfExists {
+			// Noop.
+			return &valuesNode{}, nil
+		}
+		// Key does not exist, but we want it to: error out.
+		return nil, fmt.Errorf("table %q does not exist", n.Name.Table())
+	}
+
+	if !dbDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
+		return nil, fmt.Errorf("user %s does not have %s privilege on database %s",
+			p.user, parser.PrivilegeWrite, dbDesc.Name)
+	}
+
+	if n.Name.Table() == string(n.NewName) {
```

Comment:
we should stay consistent with RenameDatabase where the noop is checked before looking up the database, meaning that `rename a to a` is a noop even if `a` does not exist.
I'm not certain this is the right behavior, but let's at least make it the same in both places.